src/groupByKey.ts

```diff
@@ -2,6 +2,23 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+interface Item {
+  [key: string]: any;
+}
+
+export function groupByKey(items: Item[], key :keyof GroupsMap<Item>):
+GroupsMap<Item> {
```

Comment:
```suggestion
export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
```

Reply:
You can use [generic](https://www.typescriptlang.org/docs/handbook/2/generics.html#generic-types) here